#### Import PySpark and other Libraries

In [0]:
import pandas as pd
from pyspark.sql import SparkSession
from pyspark.context import SparkContext
from pyspark.sql.functions import *
from pyspark.sql.types import *
from datetime import date, timedelta, datetime
import time

#### Create Spark Session

In [0]:
sc = SparkSession.builder.appName("AnalyzingIoTData").config("spark.sql.shuffle.partitions", "50").config("spark.driver.maxResultSize","5g").config("spark.sql.execution.arrow.enabled", "true").getOrCreate()

#### 1. Get the Iot Data in the Dataframe

In [0]:
#Creates a spark data frame called as raw_data.
#JSON
iot_data = sc.read.json('/FileStore/tables/iot_devices.json')

#### Display the Iot Data

In [0]:
display(iot_data)

battery_level c02_level cca2 cca3 cn device_id device_name humidity ip latitude lcd longitude scale temp timestamp 8 868 US USA United States 1 meter-gauge-1xbYRYcj 51 68.161.225.1 38.0 green -97.0 Celsius 34 1458444054093 7 1473 NO NOR Norway 2 sensor-pad-2n2Pea 70 213.161.254.1 62.47 red 6.15 Celsius 11 1458444054119 2 1556 IT ITA Italy 3 device-mac-36TWSKiT 44 88.36.5.1 42.83 red 12.83 Celsius 19 1458444054120 6 1080 US USA United States 4 sensor-pad-4mzWkz 32 66.39.173.154 44.06 yellow -121.32 Celsius 28 1458444054121 4 931 PH PHL Philippines 5 therm-stick-5gimpUrBB 62 203.82.41.9 14.58 green 120.97 Celsius 25 1458444054122 3 1210 US USA United States 6 sensor-pad-6al7RTAobR 51 204.116.105.67 35.93 yellow -85.46 Celsius 27 1458444054122 3 1129 CN CHN China 7 meter-gauge-7GeDoanM 26 220.173.179.1 22.82 yellow 108.32 Celsius 18 1458444054123 0 1536 JP JPN Japan 8 sensor-pad-8xUD6pzsQI 35 210.173.177.1 35.69 red 139.69 Celsius 27 1458444054123 3 807 JP JPN Japan 9 device-mac-9GcjZ2pw 85 118.23.68.227 35.69 green 139.69 Celsius 13 1458444054124 7 1470 US USA United States 10 sensor-pad-10BsywSYUF 56 208.109.163.218 33.61 red -111.89 Celsius 26 1458444054125 3 1544 IT ITA Italy 11 meter-gauge-11dlMTZty 85 88.213.191.34 42.83 red 12.83 Celsius 16 1458444054125 0 1260 US USA United States 12 sensor-pad-12Y2kIm0o 92 68.28.91.22 38.0 yellow -97.0 Celsius 12 1458444054126 6 1007 IN IND India 13 meter-gauge-13GrojanSGBz 92 59.144.114.250 28.6 yellow 77.2 Celsius 13 1458444054127 1 1346 NO NOR Norway 14 sensor-pad-14QL93sBR0j 90 193.156.90.200 59.95 yellow 10.75 Celsius 16 1458444054127 9 1259 US USA United States 15 device-mac-15se6mZ 70 67.185.72.1 47.41 yellow -122.0 Celsius 13 1458444054128 4 1425 US USA United States 16 sensor-pad-16aXmIJZtdO 53 68.85.85.106 38.0 red -97.0 Celsius 15 1458444054128 0 1466 US USA United States 17 meter-gauge-17zb8Fghhl 98 161.188.212.254 39.95 red -75.16 Celsius 31 1458444054129 4 1096 CN CHN China 18 sensor-pad-18XULN9Xv 25 221.3.128.242 25.04 yellow 102.72 Celsius 31 1458444054130 9 1531 US USA United States 19 meter-gauge-19eg1BpfCO 75 64.124.180.215 38.0 red -97.0 Celsius 29 1458444054130 7 1155 US USA United States 20 sensor-pad-20gFNfBgqr 33 66.153.162.66 33.94 yellow -78.92 Celsius 10 1458444054131 5 939 AT AUT Austria 21 device-mac-21sjz5h 44 193.200.142.254 48.2 green 16.37 Celsius 30 1458444054131 7 1522 JP JPN Japan 22 sensor-pad-22oWV2D 58 221.113.129.83 35.69 red 139.69 Celsius 24 1458444054132 5 1245 IN IND India 23 meter-gauge-230IupA 47 59.90.65.1 12.98 yellow 77.58 Celsius 23 1458444054133 8 1511 CA CAN Canada 24 sensor-pad-24PytzD00Cp 73 69.90.67.1 43.85 red -79.33 Celsius 28 1458444054133 4 880 US USA United States 25 therm-stick-25kK6VyzIFB 78 24.154.45.90 41.1 green -80.76 Celsius 27 1458444054134 9 1300 JP JPN Japan 26 sensor-pad-26rAyCZQOQH9 27 210.158.147.11 35.69 yellow 139.69 Celsius 10 1458444054135 5 1597 KR KOR Republic of Korea 27 device-mac-27P5wf2 73 218.239.168.1 37.57 red 126.98 Celsius 10 1458444054135 3 1502 KR KOR Republic of Korea 28 sensor-pad-28Tsudcoikw 64 211.238.224.77 37.29 red 127.01 Celsius 25 1458444054136 6 1095 NL NLD Netherlands 29 meter-gauge-29lyNVxIS 69 83.98.224.49 52.37 yellow 4.9 Celsius 15 1458444054137 6 992 US USA United States 30 sensor-pad-30FC7VCyFp0i 46 66.225.253.47 41.85 green -87.65 Celsius 14 1458444054137 5 979 ES ESP Spain 31 meter-gauge-31bcRD8TqXry 48 80.251.64.1 40.4 green -3.68 Celsius 11 1458444054138 8 1262 US USA United States 32 sensor-pad-329KFvY 30 128.83.9.146 30.3 yellow -97.74 Celsius 19 1458444054139 3 835 RU RUS Russia 33 device-mac-33B94GfPzi 30 178.23.147.134 55.75 green 37.62 Celsius 15 1458444054139 7 940 RU RUS Russia 34 sensor-pad-34F1Jubre3B 89 80.255.179.78 57.62 green 39.86 Celsius 27 1458444054140 7 1131 CN CHN China 35 therm-stick-35Lg804z 35 124.164.2.186 35.0 yellow 105.0 Celsius 21 1458444054141 1 1305 CY CYP Cyprus 36 sensor-pad-36VQv8fnEg 47 213.7.14.1 35.0 yellow 33.0 Celsius 24 1458444054141 5 881 

#### 2. Convert iot Dataframe to Temp View iot

In [0]:
iot_data.createOrReplaceTempView("iot")

#### Counting Number of Iot Devices Per Country

In [0]:
iot_device_count_by_country = spark.sql("Select CASE WHEN (cn = '') THEN 'No Country' ELSE cn END as Country, count(device_id) as Iot_Device_Count From iot group by cn order by Country")

#### 3. Displaying Counts of Iot Devices Per Country

In [0]:
display(iot_device_count_by_country)

Country Iot_Device_Count Afghanistan 19 Albania 32 Algeria 34 American Samoa 3 Andorra 4 Angola 28 Anguilla 7 Antigua and Barbuda 60 Argentina 978 Armenia 34 Aruba 8 Australia 3119 Austria 1462 Azerbaijan 49 Bahamas 18 Bahrain 55 Bangladesh 153 Barbados 38 Belarus 123 Belgium 666 Belize 13 Benin 3 Bermuda 53 Bhutan 8 Bolivia 108 Bosnia and Herzegovina 106 Botswana 12 Brazil 3224 British Indian Ocean Territory 2 British Virgin Islands 6 Brunei 19 Bulgaria 1334 Burkina Faso 24 Cambodia 43 Cameroon 29 Canada 6041 Cayman Islands 26 Chile 371 China 14455 Colombia 423 Congo 8 Cook Islands 3 Costa Rica 116 Croatia 193 Cuba 15 Curaçao 36 Cyprus 93 Czech Republic 1507 Denmark 1190 Dominica 26 Dominican Republic 24 East Timor 3 Ecuador 242 Egypt 241 El Salvador 60 Estonia 190 Falkland Islands 1 Faroe Islands 14 Federated States of Micronesia 3 Fiji 11 Finland 694 France 5305 French Polynesia 20 Gabon 1 Gambia 7 Georgia 95 Germany 7942 Ghana 30 Gibraltar 9 Greece 418 Greenland 2 Grenada 12 Guadeloupe 12 Guam 32 Guatemala 63 Guernsey 8 Guinea 1 Guyana 5 Haiti 12 Hashemite Kingdom of Jordan 46 Honduras 50 Hong Kong 1428 Hungary 772 Iceland 81 India 1867 Indonesia 811 Iran 336 Iraq 7 Ireland 446 Isle of Man 8 Israel 710 Italy 2915 Ivory Coast 10 Jamaica 44 Japan 12100 Jersey 34 Kazakhstan 228 Kenya 96 Kosovo 7 Kuwait 120 Kyrgyzstan 43 Laos 7 Latvia 358 Lebanon 72 Lesotho 5 Liberia 2 Libya 16 Liechtenstein 15 Luxembourg 88 Macao 33 Macedonia 75 Madagascar 15 Malawi 9 Malaysia 726 Maldives 11 Mali 11 Malta 63 Marshall Islands 3 Martinique 6 Mauritania 7 Mauritius 37 Mexico 1214 Monaco 12 Mongolia 39 Montenegro 14 Morocco 88 Mozambique 22 Namibia 30 Nepal 33 Netherlands 2488 New Caledonia 18 New Zealand 457 Nicaragua 47 Nigeria 58 No Country 1810 Northern Mariana Islands 9 Norway 1487 Oman 25 Pakistan 256 Palestine 25 Panama 127 Papua New Guinea 16 Paraguay 32 Peru 116 Philippines 681 Poland 2744 Portugal 450 Puerto Rico 253 Qatar 58 Republic of Korea 11879 Republic of Lithuania 312 Republic of Moldova 79 Republic of the Congo 3 Romania 1151 Russia 5989 Rwanda 6 Réunion 9 Saint Kitts and Nevis 23 Saint Lucia 6 Saint Martin 1 Saint Vincent and the Grenadines 11 Samoa 1 San Marino 3 Saudi Arabia 136 Senegal 25 Serbia 255 Seychelles 4 Sierra Leone 4 Singapore 1076 Sint Maarten 7 Slovak Republic 305 Slovenia 281 Solomon Islands 2 South Africa 560 Spain 2310 Sri Lanka 65 Sudan 32 Suriname 10 Swaziland 11 Sweden 2880 Switzerland 1321 Syria 5 Taiwan 2128 Tajikistan 4 Tanzania 53 Thailand 947 Togo 7 Tonga 1 Trinidad and Tobago 28 Tunisia 31 Turkey 681 Turkmenistan 3 Turks and Caicos Islands 1 U.S. Virgin Islands 51 Uganda 33 Ukraine 1448 United Arab Emirates 123 United Kingdom 6486 United States 68545 Uruguay 117 Uzbekistan 34 Vanuatu 10 Vatican City 1 Venezuela 190 Vietnam 246 Yemen 19 Zambia 7 Zimbabwe 27 Åland 20

* We observed there are 1810 iot devices mapped to blank country values. We have mapped them to a dummy Country named as NoCountry

#### Countries with CO2 Emission more then 1400

In [0]:
countries_with_co2_more_than_1400 = spark.sql("Select CASE WHEN (cn = '') THEN 'No Country' ELSE cn END as Country, c02_level as CO2_Level From iot WHERE c02_level>1400 ORDER BY c02_level DESC")

#### 4. Displaying  Countries with CO2 Emission more then 1400 in Descending Order

In [0]:
display(countries_with_co2_more_than_1400)

Country CO2_Level Finland 1599 United States 1599 Brazil 1599 United States 1599 Japan 1599 Japan 1599 Czech Republic 1599 United States 1599 United States 1599 United States 1599 United States 1599 China 1599 United States 1599 United States 1599 United States 1599 Australia 1599 United States 1599 United States 1599 France 1599 Bulgaria 1599 China 1599 Canada 1599 United States 1599 Latvia 1599 Germany 1599 Brazil 1599 United States 1599 United States 1599 United States 1599 Bulgaria 1599 United States 1599 United States 1599 United States 1599 Republic of Korea 1599 Japan 1599 Finland 1599 Spain 1599 Netherlands 1599 United States 1599 United States 1599 Switzerland 1599 United States 1599 Ukraine 1599 United States 1599 Russia 1599 Australia 1599 China 1599 China 1599 Mexico 1599 United States 1599 Mexico 1599 Spain 1599 Argentina 1599 Germany 1599 Russia 1599 China 1599 United States 1599 China 1599 Russia 1599 Republic of Korea 1599 United States 1599 United Kingdom 1599 Brazil 1599 Hungary 1599 Republic of Korea 1599 Japan 1599 Germany 1599 United States 1599 United States 1599 United Kingdom 1599 United States 1599 Sweden 1599 United States 1599 Germany 1599 Russia 1599 United States 1599 Germany 1599 Romania 1599 Japan 1599 Japan 1599 China 1599 Taiwan 1599 Ukraine 1599 United States 1599 Brazil 1599 United States 1599 United Kingdom 1599 Canada 1599 United States 1599 Ukraine 1599 Netherlands 1599 United States 1599 Japan 1599 United States 1599 Sweden 1599 China 1599 United States 1599 United States 1599 United States 1599 United States 1599 Netherlands 1599 Spain 1599 India 1599 Bulgaria 1599 United States 1599 United States 1599 Bulgaria 1599 United States 1599 Netherlands 1599 Japan 1599 Spain 1599 China 1599 United Kingdom 1599 United States 1599 United States 1599 Australia 1599 South Africa 1599 China 1599 United States 1599 Germany 1599 United States 1599 United States 1599 Malaysia 1599 China 1599 Romania 1599 United States 1599 United States 1599 Brazil 1599 Republic of Korea 1599 China 1599 Republic of Korea 1599 Canada 1599 United States 1599 Republic of Korea 1599 China 1599 Germany 1599 United States 1599 United States 1599 Czech Republic 1599 United Arab Emirates 1599 Japan 1599 Germany 1599 United States 1599 Hong Kong 1599 Hungary 1599 Canada 1599 United States 1599 Canada 1599 Czech Republic 1599 Turkey 1599 Philippines 1599 United States 1599 Australia 1599 United States 1599 China 1599 Australia 1599 Netherlands 1599 United States 1599 United States 1599 Russia 1599 Brazil 1599 Japan 1599 United States 1599 Japan 1599 Sweden 1599 China 1599 United States 1599 Russia 1599 United States 1599 United States 1599 United States 1599 Japan 1599 Germany 1599 China 1599 Poland 1599 United States 1599 Taiwan 1599 Canada 1599 United States 1599 Ukraine 1599 Republic of Korea 1599 Bulgaria 1599 Philippines 1599 United States 1599 United Kingdom 1599 Saint Kitts and Nevis 1599 United States 1599 United States 1599 France 1599 United States 1599 Sweden 1599 United States 1599 Republic of Korea 1599 Japan 1599 France 1599 Thailand 1599 Romania 1599 Japan 1599 Kazakhstan 1599 United States 1599 United States 1599 South Africa 1599 United States 1599 United States 1599 France 1599 China 1599 Republic of Korea 1599 United States 1599 Peru 1599 Republic of Korea 1599 United States 1599 Czech Republic 1599 Republic of Korea 1599 Republic of Korea 1599 Turkey 1599 United States 1599 United Kingdom 1599 Vietnam 1599 United States 1599 Taiwan 1599 Taiwan 1599 Hong Kong 1599 No Country 1599 United Kingdom 1599 United States 1599 Taiwan 1599 United States 1599 China 1599 Ireland 1599 Japan 1599 Bermuda 1599 China 1599 China 1599 Denmark 1599 Poland 1599 United States 1599 United States 1599 United States 1599 Japan 1599 France 1599 Italy 1599 United States 1599 Republic of Korea 1599 Egypt 1599 Japan 1599 Latvia 1599 Republic of Korea 1599 Russia 1599 Germany 1599 Egypt 1599 Bolivia 1599 United States 1599 Republic of Kor

#### Countries With High Levels of C02 With Devices

In [0]:
countries_with_highco2 = spark.sql("Select CASE WHEN (cca3 = '') THEN 'No Country' ELSE cca3 END as Country, device_id From iot WHERE LCD='red' ORDER BY Country, device_id")

#### 5. Display Countries With High Levels of C02 With Devices

In [0]:
display(countries_with_highco2)

Country device_id ABW 3251 ABW 196547 AFG 122437 AFG 131489 AFG 136829 AFG 172471 AFG 191633 AFG 197785 AGO 24240 AGO 30228 AGO 33254 AGO 40274 AGO 65166 AGO 86009 AGO 166577 AGO 187756 ALA 37695 ALA 64390 ALA 104561 ALA 162539 ALA 192095 ALB 12532 ALB 31801 ALB 82389 ALB 88955 ALB 118964 ALB 150378 AND 51399 ARE 3170 ARE 15836 ARE 17261 ARE 18198 ARE 27479 ARE 49762 ARE 50698 ARE 66716 ARE 74301 ARE 76271 ARE 82970 ARE 88390 ARE 101854 ARE 124953 ARE 131883 ARE 132621 ARE 133860 ARE 137869 ARE 148884 ARE 152485 ARE 162938 ARE 167876 ARE 174548 ARE 179391 ARE 180298 ARE 185946 ARE 196753 ARG 1198 ARG 4600 ARG 4866 ARG 5169 ARG 5514 ARG 5643 ARG 6323 ARG 6486 ARG 6692 ARG 6808 ARG 7404 ARG 8939 ARG 9985 ARG 10521 ARG 10725 ARG 11048 ARG 11553 ARG 12423 ARG 13097 ARG 13155 ARG 13207 ARG 13395 ARG 13420 ARG 13503 ARG 15149 ARG 16176 ARG 18982 ARG 19153 ARG 20004 ARG 22264 ARG 22339 ARG 22386 ARG 22410 ARG 22726 ARG 23003 ARG 24789 ARG 25004 ARG 25666 ARG 26750 ARG 27269 ARG 28052 ARG 28739 ARG 29032 ARG 29903 ARG 30341 ARG 30398 ARG 30597 ARG 30913 ARG 32735 ARG 33071 ARG 33298 ARG 33521 ARG 34226 ARG 35696 ARG 35843 ARG 36793 ARG 37381 ARG 37467 ARG 39977 ARG 40176 ARG 41091 ARG 41105 ARG 42075 ARG 43309 ARG 46942 ARG 47416 ARG 50210 ARG 51150 ARG 51244 ARG 54691 ARG 55408 ARG 57410 ARG 59718 ARG 60587 ARG 60922 ARG 61283 ARG 62571 ARG 62885 ARG 64340 ARG 64977 ARG 65632 ARG 66948 ARG 66959 ARG 67314 ARG 67716 ARG 68072 ARG 68605 ARG 68698 ARG 69804 ARG 71000 ARG 71031 ARG 71982 ARG 72216 ARG 72230 ARG 72289 ARG 72942 ARG 73602 ARG 73664 ARG 74713 ARG 76594 ARG 77626 ARG 78025 ARG 79030 ARG 79513 ARG 80823 ARG 80865 ARG 81322 ARG 83878 ARG 84197 ARG 84945 ARG 85266 ARG 85643 ARG 85732 ARG 86215 ARG 87150 ARG 88577 ARG 89519 ARG 90463 ARG 94448 ARG 95086 ARG 96014 ARG 96861 ARG 97064 ARG 97933 ARG 98846 ARG 99070 ARG 103246 ARG 103742 ARG 106650 ARG 106890 ARG 106902 ARG 107887 ARG 109776 ARG 111524 ARG 111746 ARG 111934 ARG 112214 ARG 112261 ARG 112391 ARG 114937 ARG 116315 ARG 116859 ARG 116943 ARG 120666 ARG 120969 ARG 121129 ARG 123598 ARG 123645 ARG 124391 ARG 126842 ARG 127113 ARG 127942 ARG 128706 ARG 130838 ARG 130897 ARG 132192 ARG 133268 ARG 135501 ARG 136205 ARG 136647 ARG 137101 ARG 137190 ARG 137927 ARG 138616 ARG 141618 ARG 141868 ARG 143077 ARG 143949 ARG 146580 ARG 149533 ARG 149892 ARG 150743 ARG 150991 ARG 153434 ARG 153679 ARG 153731 ARG 154258 ARG 154389 ARG 154659 ARG 154980 ARG 154983 ARG 155556 ARG 155563 ARG 155669 ARG 156039 ARG 156129 ARG 156136 ARG 156160 ARG 156454 ARG 156924 ARG 158683 ARG 159022 ARG 161860 ARG 162995 ARG 164291 ARG 164446 ARG 165340 ARG 166213 ARG 167041 ARG 168686 ARG 169447 ARG 170119 ARG 173819 ARG 173866 ARG 174073 ARG 177407 ARG 177594 ARG 179575 ARG 180672 ARG 181236 ARG 182098 ARG 184778 ARG 186034 ARG 186099 ARG 187066 ARG 187882 ARG 190033 ARG 190449 ARG 190564 ARG 190816 ARG 191530 ARG 193476 ARG 197905 ARM 28280 ARM 33002 ARM 50716 ARM 53769 ARM 73375 ARM 124405 ARM 130623 ARM 183223 ATG 15819 ATG 21780 ATG 28478 ATG 80683 ATG 88827 ATG 101025 ATG 109889 ATG 116768 ATG 119335 ATG 135169 ATG 155473 ATG 187803 ATG 187889 ATG 197248 AUS 441 AUS 939 AUS 965 AUS 972 AUS 1300 AUS 1737 AUS 1863 AUS 2624 AUS 2769 AUS 3621 AUS 3686 AUS 3845 AUS 3865 AUS 4014 AUS 4053 AUS 4183 AUS 4261 AUS 4434 AUS 4859 AUS 5149 AUS 5201 AUS 5474 AUS 5487 AUS 5882 AUS 6475 AUS 6549 AUS 6676 AUS 7391 AUS 7421 AUS 7505 AUS 7655 AUS 7667 AUS 7750 AUS 7939 AUS 8124 AUS 8391 AUS 8868 AUS 9112 AUS 9258 AUS 9749 AUS 9877 AUS 9904 AUS 10008 AUS 11100 AUS 11569 AUS 12107 AUS 12892 AUS 13027 AUS 13565 AUS 14735 AUS 14897 AUS 14913 AUS 15127 AUS 15287 AUS 16303 AUS 16429 AUS 16641 AUS 17306 AUS 17799 AUS 17857 AUS 17958 AUS 18023 AUS 18059 AUS 18245 AUS 18678 AUS 18847 AUS 19020 AUS 19532 AUS 19702 AUS 19723 AUS 20102 AUS 20522 AUS 20616 AUS 20827 AUS 21072 AUS 21126 AUS 21159 AUS 21345 AUS 21380 AUS 21397 AUS 21410 AUS 21825 AUS 22105 AUS 22216 AUS 22404 AUS 22723 AUS 22740 AUS 22963 AUS 23330 AUS 23979 AUS 2

#### Devices Which Need Battery Replacement

In [0]:
## Assuming the battery level=0 as replacement needed battery
devices_which_need_battery_replacement = spark.sql("Select CASE WHEN (cca3 = '') THEN 'No Country' ELSE cca3 END as Country, device_id From iot WHERE battery_level=0 ORDER BY cca3, device_id")

#### 6. Display Devices Which Need Battery Replacement

In [0]:
display(devices_which_need_battery_replacement)

Country device_id AFG 122629 AFG 162516 AFG 167828 AGO 65166 AIA 122109 AIA 146980 ALA 96404 ALA 124665 ALA 197516 ALB 12532 ALB 82389 ALB 138223 ARE 6353 ARE 32595 ARE 62126 ARE 64153 ARE 66197 ARE 74301 ARE 77008 ARE 81355 ARE 82970 ARE 107431 ARE 110925 ARE 123314 ARE 125184 ARE 128905 ARE 131514 ARE 162938 ARE 193916 ARG 227 ARG 943 ARG 2817 ARG 3821 ARG 6669 ARG 6692 ARG 8208 ARG 10521 ARG 10844 ARG 12363 ARG 13480 ARG 13503 ARG 14484 ARG 17558 ARG 18134 ARG 18533 ARG 21252 ARG 22735 ARG 25749 ARG 30913 ARG 33071 ARG 33547 ARG 34171 ARG 42703 ARG 43040 ARG 43531 ARG 47456 ARG 47657 ARG 48226 ARG 51244 ARG 51805 ARG 56501 ARG 59718 ARG 60369 ARG 61911 ARG 68241 ARG 68339 ARG 69804 ARG 70521 ARG 70962 ARG 72311 ARG 74170 ARG 74889 ARG 75119 ARG 75272 ARG 75378 ARG 76017 ARG 76936 ARG 79551 ARG 79643 ARG 80823 ARG 81532 ARG 82271 ARG 83341 ARG 86755 ARG 89519 ARG 91800 ARG 94283 ARG 96366 ARG 96861 ARG 98165 ARG 98846 ARG 99070 ARG 103022 ARG 105879 ARG 106556 ARG 107184 ARG 109549 ARG 110703 ARG 111035 ARG 112214 ARG 112391 ARG 114356 ARG 114716 ARG 116034 ARG 117675 ARG 123552 ARG 124084 ARG 126764 ARG 127113 ARG 128806 ARG 130897 ARG 131511 ARG 132915 ARG 134323 ARG 135444 ARG 137101 ARG 138616 ARG 138971 ARG 139059 ARG 141489 ARG 143949 ARG 152395 ARG 154389 ARG 154841 ARG 154983 ARG 155254 ARG 155320 ARG 156129 ARG 157151 ARG 158153 ARG 159022 ARG 159601 ARG 161099 ARG 162789 ARG 164291 ARG 164594 ARG 166650 ARG 167486 ARG 169808 ARG 170525 ARG 172955 ARG 173566 ARG 174560 ARG 175760 ARG 176075 ARG 183553 ARG 184623 ARG 185191 ARG 187653 ARG 187655 ARG 189169 ARG 189307 ARG 190069 ARG 190564 ARG 194809 ARG 195587 ARG 197266 ARG 197355 ARM 28280 ARM 124100 ATG 88827 ATG 144290 ATG 197248 AUS 111 AUS 439 AUS 817 AUS 939 AUS 2742 AUS 2788 AUS 3845 AUS 6339 AUS 6497 AUS 6890 AUS 6998 AUS 7391 AUS 7505 AUS 7931 AUS 8124 AUS 8997 AUS 10736 AUS 11347 AUS 12282 AUS 12500 AUS 12889 AUS 14202 AUS 14557 AUS 15221 AUS 16528 AUS 17914 AUS 18059 AUS 18565 AUS 18629 AUS 19723 AUS 20578 AUS 22603 AUS 25020 AUS 26160 AUS 26272 AUS 26705 AUS 26970 AUS 27090 AUS 27252 AUS 29205 AUS 31043 AUS 32127 AUS 32961 AUS 33781 AUS 35269 AUS 37481 AUS 38216 AUS 38309 AUS 38548 AUS 38676 AUS 40162 AUS 40343 AUS 40652 AUS 40791 AUS 40888 AUS 41008 AUS 41519 AUS 41927 AUS 41973 AUS 42519 AUS 42955 AUS 43191 AUS 43308 AUS 44373 AUS 44476 AUS 45204 AUS 45745 AUS 47027 AUS 47419 AUS 47601 AUS 47655 AUS 47687 AUS 48335 AUS 49203 AUS 49752 AUS 52249 AUS 52374 AUS 53303 AUS 54063 AUS 54410 AUS 55081 AUS 55379 AUS 56252 AUS 56857 AUS 57244 AUS 57767 AUS 57786 AUS 58318 AUS 58368 AUS 58937 AUS 58961 AUS 59921 AUS 60661 AUS 61098 AUS 62139 AUS 63345 AUS 63440 AUS 63947 AUS 65142 AUS 65682 AUS 66218 AUS 66477 AUS 68758 AUS 69391 AUS 69565 AUS 69733 AUS 69739 AUS 70219 AUS 70634 AUS 71374 AUS 71397 AUS 72245 AUS 72406 AUS 72723 AUS 73138 AUS 74435 AUS 74550 AUS 76660 AUS 76728 AUS 77485 AUS 77880 AUS 79311 AUS 82519 AUS 82854 AUS 82965 AUS 83105 AUS 85247 AUS 85432 AUS 85756 AUS 85770 AUS 86197 AUS 86324 AUS 86650 AUS 88024 AUS 88582 AUS 89056 AUS 91214 AUS 91252 AUS 91371 AUS 91411 AUS 92089 AUS 92567 AUS 92620 AUS 92736 AUS 93099 AUS 93434 AUS 93834 AUS 94939 AUS 95299 AUS 95359 AUS 95411 AUS 96487 AUS 97612 AUS 97862 AUS 97948 AUS 98588 AUS 98823 AUS 99521 AUS 99694 AUS 100376 AUS 100741 AUS 100882 AUS 101116 AUS 101499 AUS 102467 AUS 103037 AUS 103178 AUS 103825 AUS 104137 AUS 104705 AUS 106061 AUS 106067 AUS 106479 AUS 106597 AUS 106760 AUS 108137 AUS 108542 AUS 108558 AUS 108801 AUS 109157 AUS 109596 AUS 110075 AUS 110745 AUS 111036 AUS 111145 AUS 113156 AUS 113592 AUS 114114 AUS 115035 AUS 115214 AUS 115253 AUS 115426 AUS 115898 AUS 117300 AUS 118059 AUS 118413 AUS 119387 AUS 121035 AUS 121529 AUS 122656 AUS 123661 AUS 123709 AUS 123779 AUS 124255 AUS 124407 AUS 125458 AUS 125501 AUS 125573 AUS 125641 AUS 127435 AUS 127722 AUS 128040 AUS 128065 AUS 130032 AUS 132406 AUS 133445 AUS 133694 AUS 133715 AUS 134838 AUS 137256 AUS 137424 AUS 138487 AUS 138634 AUS